app/services/participants/change_relationship.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module Participants
+  class ChangeRelationship < BaseService
+    def call
+      return if already_with_same_partnership?
+
+      check_partnership_is_suitable!
+
+      ActiveRecord::Base.transaction do
+        create_induction_programme! if induction_programme.blank?
+
+        if fixing_mistake
+          change_for_mistake!
+        else
+          change_for_circumstance!
+        end
+      end
+    end
+
+  private
+
+    attr_reader :induction_record, :partnership, :fixing_mistake
+
+    def initialize(induction_record:, partnership:, fixing_mistake: false)
+      @induction_record = induction_record
+      @partnership = partnership
+      @fixing_mistake = fixing_mistake
+    end
+
+    def change_for_mistake!
+      check_participant_does_not_have_declarations_with_current_provider!
+      old_induction_programme = induction_record.induction_programme
+
+      induction_record.update!(induction_programme:)
+
+      remove_old_induction_programme_if_possible!(old_induction_programme:)
+    end
+
+    def change_for_circumstance!
+      Induction::ChangeInductionRecord.call(induction_record:, changes: { induction_programme: })
+    end
+
+    def check_participant_does_not_have_declarations_with_current_provider!
+      raise ArgumentError, "Participant has declarations with current provider!" if has_declarations_with_current_provider?
+    end
+
+    def has_declarations_with_current_provider?
+      ParticipantDeclaration::ECF.where(participant_profile:,
+                                        cpd_lead_provider:).any?
+    end
+
+    def check_partnership_is_suitable!
+      if partnership_in_a_different_cohort?
```

Comment:
Im not sure but does a partnership being pending true have any implications on the logic, can't seem to find where we would set it as true though 🤔

Reply:
I don't think `pending` is used anywhere, it seems we create partnerships with it set to `false` in the services?